core/src/com/unciv/ui/screens/cityscreen/ConstructionInfoTable.kt

```diff
@@ -99,6 +100,18 @@ class ConstructionInfoTable(val cityScreen: CityScreen) : Table() {
                     selectedConstructionTable.add(button).padTop(5f).colspan(2).center().row()
                 }
             }
+            if (construction is BaseUnit) {
+                val unitType = construction.unitType
+                
+                if (city.canBuildUnitTypeWithSavedPromotion[unitType] != null) {
+                    row()
+                    add(city.canBuildUnitTypeWithSavedPromotion[unitType]?.let {
+                        "Default promotions".toCheckBox(
```

Comment:
Do not use .let here, instead use regular assignment
If this is bull then you'd get add(null) which is not what we want, probably